File: SCADA/webapi/Model/Alarm.cs
using System.Collections.ObjectModel;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Net;
using webapi.Enum;

namespace webapi.model
{

    public class Alarm
    {
        [Key]
        public int Id { get; set; }
        public AlarmType Type {get; set;}
        public AlarmPriority Priority { get; set; }
        public double Limit { get; set; }
        public AnalogInput AnalogInput { get; set; }

    }
}

File: SCADA/angularapp/src/app/models/alarm.ts
import { AnalogInput } from "./tags";

export interface Alarm {
    id: number;
    type: AlarmType;
    priority: AlarmPriority;
    limit: number;
    analogInput: AnalogInput;
    date: Date;
  }
  
  export enum AlarmPriority {
    HIGH_PRIORITY = 'high',
    NORMAL_PRIORITY = 'normal',
    LOW_PRIORITY = 'low'
  }
  
  export enum AlarmType {
    LOW = 'low',
    HIGH = 'high'
  }